azurerm/data_source_container_orchestrator_versions.go

```diff
@@ -0,0 +1,78 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/go-version"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func dataSourceArmKubernetesServiceVersions() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceArmKubernetesServiceVersionsRead,
+
+		Schema: map[string]*schema.Schema{
+			"location": locationSchema(),
+			"version_prefix": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"versions": {
+				Type:     schema.TypeList,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Computed: true,
+			},
+			"latest_version": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceArmKubernetesServiceVersionsRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).containerServicesClient
+	ctx := meta.(*ArmClient).StopContext
+
+	location := d.Get("location").(string)
```

Comment:
we probably want to run this through `azurermNormalizeLocation(location)` to ensure this value's normalized?